conans/client/build/cmake.py

```diff
@@ -1,27 +1,20 @@
 import os
 import platform
-import subprocess
-from itertools import chain
-
-from six import StringIO  # Python 2 and 3 compatible
 
 from conans.client import tools
 from conans.client.build import defs_to_string, join_arguments
+from conans.client.build.base_cmake import BaseCMake, _compute_build_flags
 from conans.client.build.cmake_flags import CMakeDefinitionsBuilder, \
     get_generator, is_multi_configuration, verbose_definition, verbose_definition_name, \
     cmake_install_prefix_var_name, get_toolset, build_type_definition, \
-    cmake_in_local_cache_var_name, runtime_definition_var_name, get_generator_platform, \
+    runtime_definition_var_name, get_generator_platform, \
     is_generator_platform_supported, is_toolset_supported
-from conans.client.output import ConanOutput
-from conans.client.tools.oss import cpu_count, args_to_string
 from conans.errors import ConanException
-from conans.model.conan_file import ConanFile
-from conans.model.version import Version
 from conans.util.config_parser import get_bool_from_text
-from conans.util.files import mkdir, get_abs_path, walk, decode_text
+from conans.util.files import get_abs_path
 
 
-class CMake(object):
+class CMake(BaseCMake):
```

Comment:
Even if this is the conceptually correct way to do it, the risk of breaking something is just too high.
Furthermore, I think the new build helper that will use toolchains, is a great opportunity to get rid of many of the legacy things in the existing one, so starting an alternative one that doesn't share a common base class might be much easier to evolve, debug, while making sure we are not breaking users at every step.

Reply:
The intention was to move to `BaseCMake` only the shared functionality, and there is a lot of things in common. I understand the risk, but I will vote depending on the road ahead: if we are going to deprecate current `CMake` helper for Conan 2.0 then I agree with the fork/duplication/divergence of the two classes.

Reply:
Done